fe/fe-core/src/main/java/com/starrocks/sql/optimizer/property/ValueProperty.java

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.property;
+
+import com.google.common.collect.Maps;
+import com.starrocks.sql.optimizer.base.ColumnRefSet;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public class ValueProperty {
+
+    private Map<ScalarOperator, ValueWrapper> valueMap;
```

Comment:
What't the meaning of `valueMap`? A ScalarOperator and its associated RangeDescriptor?

Reply:
yes. key is the scalarOperator, value is its valueinfo including a predicate associated with it and its rangeinfo.